downstairs/src/extent_inner_raw.rs

```diff
@@ -149,38 +149,172 @@ impl ExtentInner for RawInner {
         Ok(self.dirty)
     }
 
+    /// Performs a single write within this extent
     fn write(
         &mut self,
         job_id: JobId,
-        writes: &[crucible_protocol::Write],
+        write: &ExtentWrite,
         only_write_unwritten: bool,
-        iov_max: usize,
+        _iov_max: usize,
     ) -> Result<(), CrucibleError> {
-        // If the same block is written multiple times in a single write, then
-        // (1) that's weird, and (2) we need to handle it specially.  To handle
-        // such cases, we split the `writes` slice into sub-slices of unique
-        // writes.
-        let mut seen = HashSet::new();
-        let mut start = 0;
-        for i in 0..=writes.len() {
-            // If this value is a duplicate or we have reached the end of
-            // the list, then write everything up to this point and adjust
-            // our starting point and `seen` array
-            if i == writes.len() || !seen.insert(writes[i].offset.value) {
-                self.write_without_overlaps(
-                    job_id,
-                    &writes[start..i],
-                    only_write_unwritten,
-                    iov_max,
-                )?;
-                // Keep going, resetting the hashmap and start position
-                if i != writes.len() {
-                    seen.clear();
-                    seen.insert(writes[i].offset.value);
-                    start = i;
+        check_input(self.extent_size, write.offset, &write.data)?;
```

Comment:
This diff is transplanting the old `write_without_overlaps` into `write`, because that's now guaranteed by the API.